packages/grafana-ui/src/components/Graph/utils.ts

```diff
@@ -119,19 +119,45 @@ export const graphTimeFormat = (ticks: number | null, min: number | null, max: n
     const oneYear = 31536000000;
 
     if (secPerTick <= 45) {
-      return 'HH:mm:ss';
+      return localTimeFormat({ hour: '2-digit', minute: '2-digit', second: '2-digit', hour12: false }, 'HH:mm:ss');
     }
     if (secPerTick <= 7200 || range <= oneDay) {
-      return 'HH:mm';
+      return localTimeFormat({ hour: '2-digit', minute: '2-digit', hour12: false }, 'HH:mm');
     }
     if (secPerTick <= 80000) {
-      return 'MM/DD HH:mm';
+      return localTimeFormat(
+        { month: '2-digit', day: '2-digit', hour: '2-digit', minute: '2-digit', hour12: false },
+        'MM/DD HH:mm'
+      );
     }
     if (secPerTick <= 2419200 || range <= oneYear) {
-      return 'MM/DD';
+      return localTimeFormat({ month: '2-digit', day: '2-digit', hour12: false }, 'MM/DD');
     }
-    return 'YYYY-MM';
+    return localTimeFormat({ year: 'numeric', month: '2-digit', hour12: false }, 'YYYY-MM');
   }
 
-  return 'HH:mm';
+  return localTimeFormat({ hour: '2-digit', minute: '2-digit', hour12: false }, 'HH:mm');
+};
+
+const localTimeFormat = (options: Intl.DateTimeFormatOptions, fallback: string): string => {
```

Comment:
I think this should belong to https://github.com/grafana/grafana/blob/1a0c1a39e4974220890f6a8d003e545847c15863/packages/grafana-data/src/valueFormats/dateTimeFormatters.ts

Also, as @hugohaggmark mentioned, some tests for this are more than welcome :)

Reply:
Hi!
I can move if you think that is better.
From my point of view `localTimeFormat` is different from other functions in `dateTimeFormatters.ts`

`localTimeFormat` returns date format, rather than formatted date. But I can move if you say so.